core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,153 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
     test_can_play_entity.__wrapped__ = True
 
     return test_can_play_entity
+
+
+def can_view_voiceover_applications(handler):
+    """Decorator to check whether the user can access voiceover applications for
+    a given purpose.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to access voiceover application for a given
+            purpose.
+    """
+
+    def test_can_view_voiceover_applications(self, purpose, **kwargs):
```

Comment:
That's odd -- I thought I commented on this somewhere. But can you please remove the "purpose" part of the URL as we discussed, and make the handlers more "stateless"?

Reply:
Created separate handlers for each purpose.